src/upstreams.go

```diff
@@ -33,19 +34,22 @@ func ReverseProxyUpstreamList(w http.ResponseWriter, r *http.Request) {
 
 	activeUpstreams := targetEndpoint.ActiveOrigins
 	inactiveUpstreams := targetEndpoint.InactiveOrigins
-	// Sort the upstreams slice by weight, then by origin domain alphabetically
-	sort.Slice(activeUpstreams, func(i, j int) bool {
-		if activeUpstreams[i].Weight != activeUpstreams[j].Weight {
-			return activeUpstreams[i].Weight > activeUpstreams[j].Weight
+	slices.SortFunc(activeUpstreams, func(i, j *loadbalance.Upstream) int {
```

Comment:
This has a similar problem as well, the slices package was not introduce in Go 1.20 so shall not be used in Zoraxy v3.

Reply:
Ok, I'll taking care of compatible with go 1.20 and will let you know when it's done.
ps.
ps. I'm working on random upstreams func should return immediately if only just 1 or empty upstream, could be included in this PR or should I open new PR ?

Reply:
You can have both of them in one PR, I can review them at the same time as soon as you have well documented comments that shows what does what in the code.

Reply:
@tobychui Ok, ready for review.